### PR TITLE
Extract TableLayoutSettings classes

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Extract TableLayoutSettings classes

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2562)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/TableLayoutSettings.TableLayoutSettingsStub.cs [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                containerInfo.RowStyles = rowStyles;
+                containerInfo.ColumnStyles = columnStyles;
+
+                // since we've given over the styles to the other guy, null out.
+                columnStyles = null;
+                rowStyles = null;
+
+                // set a flag for assertion detection.
+                isValid = false;
+
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/TableLayoutSettings.TableLayoutSettingsStub.cs]: Please remove the extra line

File: src/System.Windows.Forms/src/System/Windows/Forms/TableLayoutSettings.TableLayoutSettingsStub.cs [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            public void SetColumn(object controlName, int column)
+            {
+                if (GetColumn(controlName) != column)
+                {
+                    ControlInformation info = GetControlInformation(controlName);
+                    info.Column = column;
+                    SetControlInformation(controlName, info);
+                }
+
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/TableLayoutSettings.TableLayoutSettingsStub.cs]: please remove the extra line

File: src/System.Windows.Forms/src/System/Windows/Forms/TableLayoutSettings.TableLayoutSettingsStub.cs [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+            }
+            public void SetColumnSpan(object controlName, int value)
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/TableLayoutSettings.TableLayoutSettingsStub.cs]: please add a line between functions

File: src/System.Windows.Forms/src/System/Windows/Forms/TableLayoutSettings.TableLayoutSettingsStub.cs [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+            {
+                if (GetColumnSpan(controlName) != value)
+                {
+                    ControlInformation info = GetControlInformation(controlName);
+                    info.ColumnSpan = value;
+                    SetControlInformation(controlName, info);
+                }
+            }
+            public void SetRow(object controlName, int row)
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/TableLayoutSettings.TableLayoutSettingsStub.cs]: please add an empty line between the functions

File: src/System.Windows.Forms/src/System/Windows/Forms/TableLayoutSettings.TableLayoutSettingsStub.cs [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+            {
+                if (GetRow(controlName) != row)
+                {
+                    ControlInformation info = GetControlInformation(controlName);
+                    info.Row = row;
+                    SetControlInformation(controlName, info);
+                }
+            }
+            public void SetRowSpan(object controlName, int value)
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/TableLayoutSettings.TableLayoutSettingsStub.cs]: please add an empty line between the functions

File: src/System.Windows.Forms/src/System/Windows/Forms/TableLayoutSettings.TableLayoutSettingsStub.cs [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+            {
+                if (GetRowSpan(controlName) != value)
+                {
+                    ControlInformation info = GetControlInformation(controlName);
+                    info.RowSpan = value;
+                    SetControlInformation(controlName, info);
+                }
+            }
+
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/TableLayoutSettings.TableLayoutSettingsStub.cs]: please remove the extra line

File: src/System.Windows.Forms/src/System/Windows/Forms/TableLayoutSettings.cs [hunk 7]
```diff
@@ -15,7 +15,7 @@ namespace System.Windows.Forms
     /// <summary>this is a wrapper class to expose interesting properties of TableLayout</summary>
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/TableLayoutSettings.cs]: nit: split this comment into 3 lines

**Comment by hughbe** [comment]
I've now performed a bit of code cleanup including renaming fields, removing new lines, using expression member bodies and adding new lines

File: src/System.Windows.Forms/src/System/Windows/Forms/TableLayoutSettings.TableLayoutSettingsStub.cs [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+                {
+                    return s_defaultControlInfo;
+                }
+                if (!_controlsInfo.ContainsKey(controlName))
+                {
+                    return s_defaultControlInfo;
+                }
+
+                return _controlsInfo[controlName];
+
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/TableLayoutSettings.TableLayoutSettingsStub.cs]: extra line after the `return` statement

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/TableLayoutSettings.TableLayoutSettingsStub.cs]: fixed

**Pull request merged**